beacon-chain/state/state-native/v1/getters_state.go

```diff
@@ -1,115 +1,188 @@
+		}
+	}
+	var hRoots [][]byte
+	if b.historicalRoots != nil {
+		hRoots = make([][]byte, len(b.historicalRoots))
+		for i, r := range b.historicalRoots {
+			tmp := r
+			hRoots[i] = tmp[:]
+		}
+	}
+	var mixes [][]byte
+	if b.randaoMixes != nil {
+		mixes = make([][]byte, len(b.randaoMixes))
+		for i, m := range b.randaoMixes {
+			tmp := m
+			mixes[i] = tmp[:]
+		}
+	}
+
+	return &ethpb.BeaconState{
+		GenesisTime:                 b.genesisTime,
+		GenesisValidatorsRoot:       b.genesisValidatorsRoot[:],
+		Slot:                        b.slot,
+		Fork:                        b.fork,
+		LatestBlockHeader:           b.latestBlockHeader,
+		BlockRoots:                  bRoots,
+		StateRoots:                  sRoots,
+		HistoricalRoots:             hRoots,
+		Eth1Data:                    b.eth1Data,
+		Eth1DataVotes:               b.eth1DataVotes,
+		Eth1DepositIndex:            b.eth1DepositIndex,
+		Validators:                  b.validators,
+		Balances:                    b.balances,
+		RandaoMixes:                 mixes,
+		Slashings:                   b.slashings,
+		PreviousEpochAttestations:   b.previousEpochAttestations,
+		CurrentEpochAttestations:    b.currentEpochAttestations,
+		JustificationBits:           b.justificationBits,
+		PreviousJustifiedCheckpoint: b.previousJustifiedCheckpoint,
+		CurrentJustifiedCheckpoint:  b.currentJustifiedCheckpoint,
+		FinalizedCheckpoint:         b.finalizedCheckpoint,
+	}
 }
 
-// CloneInnerState the beacon state into a protobuf for usage.
-func (b *BeaconState) CloneInnerState() interface{} {
-	if b == nil || b.state == nil {
+// ToProto the beacon state into a protobuf for usage.
+func (b *BeaconState) ToProto() interface{} {
+	if b == nil {
 		return nil
 	}
 
 	b.lock.RLock()
 	defer b.lock.RUnlock()
-	return &ethpb.BeaconState{
-		GenesisTime:                 b.genesisTime(),
-		GenesisValidatorsRoot:       b.genesisValidatorRoot(),
-		Slot:                        b.slot(),
-		Fork:                        b.fork(),
-		LatestBlockHeader:           b.latestBlockHeader(),
-		BlockRoots:                  b.blockRoots(),
-		StateRoots:                  b.stateRoots(),
-		HistoricalRoots:             b.historicalRoots(),
-		Eth1Data:                    b.eth1Data(),
-		Eth1DataVotes:               b.eth1DataVotes(),
-		Eth1DepositIndex:            b.eth1DepositIndex(),
-		Validators:                  b.validators(),
-		Balances:                    b.balances(),
-		RandaoMixes:                 b.randaoMixes(),
-		Slashings:                   b.slashings(),
-		PreviousEpochAttestations:   b.previousEpochAttestations(),
-		CurrentEpochAttestations:    b.currentEpochAttestations(),
-		JustificationBits:           b.justificationBits(),
-		PreviousJustifiedCheckpoint: b.previousJustifiedCheckpoint(),
-		CurrentJustifiedCheckpoint:  b.currentJustifiedCheckpoint(),
-		FinalizedCheckpoint:         b.finalizedCheckpoint(),
+
+	gvr := b.genesisValidatorsRoot
+	var bRoots [][]byte
+	if b.blockRoots != nil {
```

Comment:
same thing here, use our getters directly.

Reply:
`ToProto` obtains the state lock. Using the getter here will result in a deadlock.